python/pyspark/pandas/tests/frame/test_reshaping.py

```diff
@@ -291,7 +291,8 @@ def test_stack(self):
         psdf_multi_level_cols2 = ps.from_pandas(pdf_multi_level_cols2)
 
         self.assert_eq(
-            psdf_multi_level_cols2.stack().sort_index(), pdf_multi_level_cols2.stack().sort_index()
+            psdf_multi_level_cols2.stack().sort_index()[["weight", "height"]],
+            pdf_multi_level_cols2.stack().sort_index()[["weight", "height"]],
```

Comment:
This just for handling the column order:

**DataFrame**
```python
>>> pdf
    weight height
        kg      m
cat    1.0    2.0
dog    3.0    4.0
```

**DataFrame.stack() in Pandas 1.5.3**
```python
>>> pdf.stack()
        weight  height
cat kg     1.0     NaN
    m      NaN     2.0
dog kg     3.0     NaN
    m      NaN     4.0
```

**DataFrame.stack() in Pandas 2.1.0**
```python
>>> pdf.stack()
        weight  height
cat kg     1.0     NaN
    m      NaN     2.0
dog kg     3.0     NaN
    m      NaN     4.0
```

I think maybe this is the minor bug in Pandas, so I reported for the Pandas community to make sure.